src/Type/Php/SprintfFunctionDynamicReturnTypeExtension.php

```diff
@@ -125,32 +125,45 @@ public function getTypeFromFunctionCall(
 			return $singlePlaceholderEarlyReturn;
 		}
 
+		if ($allPatternsNonFalsy) {
+			return new IntersectionType([
+				new StringType(),
+				new AccessoryNonFalsyStringType(),
+			]);
+		}
+
 		$isNonEmpty = $allPatternsNonEmpty;
 		if (
-			count($formatStrings) === 0
+			!$isNonEmpty
 			&& $functionReflection->getName() === 'sprintf'
-			&& count($args) === 2
+			&& count($args) >= 2
```

Comment:
There was a special case for sprintf($nonEmpty, $nonEmpty, $nonEmpty) but this can be extended to `sprintf($nonEmpty, $nonEmpty, $nonEmpty)` and so on.